http/search.go

```diff
@@ -80,8 +83,22 @@ func (h *SearchHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		}
 	}
 
+	// check for timeout and create context
+	var ctx context.Context
+	timeoutStr := req.FormValue("timeout")
+	if timeoutStr == "" {
+		ctx = context.Background()
+	} else {
+		timeout, err := strconv.Atoi(timeoutStr)
+		if err != nil {
+			showError(w, req, fmt.Sprintf("error parsing timeout value: %v", err), 400)
+			return
+		}
+		ctx, _ = context.WithTimeout(context.Background(), time.Duration(timeout)*time.Millisecond)
+	}
+
 	// execute the query
-	searchResponse, err := index.Search(&searchRequest)
+	searchResponse, err := index.SearchInContext(ctx, &searchRequest)
```

Comment:
SearchInContext implementation of indexImpl and indexAliasImpl will have to adapt to this change.
They will now multiplex on ctx.Done() and result channel

Something like this:
```
select {
  case <-ctx.Done():
          fmt.Printf("timeout")

          return nil, timeoutError
  case res := <-resChan: // A goroutine will carry out the search and will update this channel with the result
           fmt.Printf("success")
           return res, nil
  }

```

Reply:
Also, To test the correctness of the implementation.
We should write a test case, where we can set a very small timeout value (few ms) and see if SearchInContext is stopped on that value or not.


All the goroutines spawned by SearchInContext ( and their children) should obey the timeout (stop the work).

Reply:
---
I see (*TopNCollector)Collect is already taking care context deadline.
```
if hc.total%CheckDoneEvery == 0 {
	select {
	case <-ctx.Done():
		search.RecordSearchCost(ctx, search.AbortM, 0)
		return ctx.Err()
	default:
	}
}
```
So, maybe SearchInContext impls don't need any change.

Reply:
Hey Shaad, I tested the implementation outside of the ServeHTTP function with a test case resembling the same implementation of using SearchInContext with a timeout. It does work as intended. But I was wondering which file this test case should go into

Reply:
There are already existing test cases for search in context where the context is created using the timeout value